coordinator/internal/logic/provertask/batch_prover_task.go

```diff
@@ -61,13 +69,42 @@ func (bp *BatchProverTask) Assign(ctx *gin.Context, getTaskParameter *coordinato
 		return nil, fmt.Errorf("check prover task parameter failed, error:%w", err)
 	}
 
+	// TODO Only happen when the first time hard fork, the ForkBlockNumber = 0.
+	// TODO need check fork block number whether equal 0 after the first hard fork
+	if getTaskParameter.ForkBlockNumber != 0 && !bp.forkMap[getTaskParameter.ForkBlockNumber] {
```

Comment:
I'm not sure I understand the logic here.

The "fork block number" should not be reported by the prover. We know the prover version (e.g. v0 or v1), and we know the hard for block numbers, so the coordinator can decide which chunks/batches to assign to which provers.

Reply:
discussed